.github/workflows/program-nft-packs.yml

```diff
@@ -0,0 +1,53 @@
+name: Program NFT Packs
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+env:
+  CARGO_TERM_COLOR: always
+  SOLANA_VERSION: 1.8.5
+  RUST_TOOLCHAIN: stable
+
+jobs:
+  build-and-test-nft-packs:
+    runs-on: ubuntu-latest
+    env:
+      cache_id: program-nft-packs 
+
+    steps:
+      # Setup Deps
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/install-linux-build-deps
+      - uses: ./.github/actions/install-solana
+        with: 
+          solana_version: ${{ env.SOLANA_VERSION }}
+      - uses: ./.github/actions/install-rust
+        with: 
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+
+      # Restore Cache from previous build/test
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ./rust/target
+          key: ${{ env.cache_id }}-${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}-${{ env.RUSTC_HASH }}
+
+      # Build deps
+      - uses: ./.github/actions/build-token-vault
+      - uses: ./.github/actions/build-metaplex
+      - uses: ./.github/actions/build-token-metadata
```

Comment:
You may not need this as when you build metaplex it'll build the other two since they're dependencies:
```
mpl-token-vault = { path = "../../token-vault/program", features = [ "no-entrypoint" ], version="0.0.1" }
mpl-token-metadata = { path = "../../token-metadata/program", features = [ "no-entrypoint" ], version="0.0.1" }
```
Therefore  all `.so` files will be produced so you could also use them for integration tests once you add those.

Building the other two separately could also mess with caching of the target folder (each action tries to cache it's result).

As a side-note I'm still not entirely sure how to cache this cleanly but we'll address this later.

Reply:
Oh ok thanks
Also have question about another one dependency - random-oracle program
NFT-packs depends on that one program but it's from different [repository](https://github.com/metaplex-foundation/randomness-oracle)
How can I build it in GH workflow as dependency ?

Reply:
We'll have to setup an action to build this somehow so you can use it in your workflow.

Basically it would entail either cloning the repo and building the program needed or somehow pulling down the `.so` file by other means.